src/Data/Singletons.hs

```diff
@@ -35,6 +35,7 @@ module Data.Singletons (
   Sing(SLambda, applySing), (@@),
 
   SingI(..), SingKind(..),
+  pattern AsSing,
```

Comment:
I would this next to `withSomeSing` in the export list, similarly to now `pattern SingI` is currently next to `singInstance`.